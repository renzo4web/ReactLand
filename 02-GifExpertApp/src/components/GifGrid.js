import PropTypes from "prop-types";

import useFetchGifs from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category);

  if (loading) return <h4>Loading</h4>;
  return (
    <ul className="grid relative animate-pulse  rounded-md border-4 bg-purple-100 border-4 border-purple-600 shadow-xl p-2">
      <h4 className="absolute -top-10 rounded-tl-lg rounded-tr-lg left-0 bg-purple-600 text-white px-3 p-2 text-xl ">
        {category}
      </h4>
      {data.map((gif) => (
        <GifGridItem key={gif.id} {...gif} />
      ))}
    </ul>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
export default GifGrid;
