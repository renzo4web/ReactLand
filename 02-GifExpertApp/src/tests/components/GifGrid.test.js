import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Test <GifGrid />", () => {
  const category = "morty";
  let wrapper = null;
  let gifs = [];
  beforeEach(() => {
    jest.clearAllMocks();
    gifs = [
      {
        id: "ABC",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "Cualquier Cosa",
      },
      {
        id: "123",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "Cualquier Cosa",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    wrapper = shallow(<GifGrid category={category} />);
  });

  test("should <GifGrid /> receibe a category", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should display items from useFetchGifs", () => {
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
