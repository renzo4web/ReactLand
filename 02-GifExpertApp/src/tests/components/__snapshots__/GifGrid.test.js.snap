// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> should <GifGrid /> receibe a category 1`] = `
<h4>
  Loading
</h4>
`;
